### PR TITLE
Fixes to autosubmit in in study view

### PR DESCRIPTION
**Pull request by alisman**

### CODE REVIEW

**Comment by alisman** [comment]
@pvannierop @BasLee this PR fixes what I think is all of the remaining issues:
1. When you toggle auto submit mode, it doesn't re-render the charts
2. The gene selection issue (missing after addition to filter)
3. Applies crawling ants to submit button
4. Submits any pending filters when user switches to auto mode (questionable but probably edge case)

File: src/pages/studyView/menu/AutosubmitToggle.tsx [hunk 1]
```diff
@@ -4,22 +4,21 @@ import classNames from 'classnames';
...
-
 @observer
 export class AutosubmitToggle extends React.Component<
     IAutosubmitToggleProps,
     {}
 > {
-    private updateHesitateMode(hesitateMode: boolean) {
-        localStorage.setItem(STUDY_VIEW_FILTER_AUTOSUBMIT, '' + hesitateMode);
-        this.context.store.hesitateUpdate = hesitateMode;
+    @autobind
```
Comment (alisman) [inline src/pages/studyView/menu/AutosubmitToggle.tsx:18]: @BasLee the idea of this refactor is that we should not overuse React context.  It's better to pass in a callback when the parent has direct access to the necessary things (the store) and there's only one (or a few) parents.

File: src/pages/studyView/menu/AutosubmitToggle.tsx [hunk 2]
```diff
@@ -45,17 +44,21 @@ export class AutosubmitToggle extends React.Component<
                 <div className="btn-group">
                     <label>
                         <input
-                            onClick={() => this.updateHesitateMode(false)}
-                            checked={!this.props.store.hesitateUpdate}
+                            onChange={this.handleChange}
+                            defaultChecked={!this.props.hesitateModeOn}
```
Comment (alisman) [inline src/pages/studyView/menu/AutosubmitToggle.tsx:48]: This was necssary so control manages its own state and we can remove observable status from hesitateModeOn (this makes it so charts don't re-render on toggle).

File: src/pages/studyView/menu/AutosubmitToggle.tsx (hunk 1 above)
Comment (BasLee) [reply src/pages/studyView/menu/AutosubmitToggle.tsx:18]: check!

File: src/pages/studyView/menu/AutosubmitToggle.tsx (hunk 2 above)
Comment (BasLee) [reply src/pages/studyView/menu/AutosubmitToggle.tsx:48]: aaa yes, that was the trick, I knew there was a better solution, thanks!

**Pull request merged**